linker/Mono.Linker.Steps/SweepStep.cs

```diff
@@ -175,6 +175,8 @@ bool ResolveAllTypeReferences (AssemblyDefinition assembly)
 			foreach (TypeReference tr in assembly.MainModule.GetTypeReferences ()) {
 				if (hash.ContainsKey (tr))
 					continue;
+				if (!CanChangeScope (tr))
```

Comment:
By "all of the logic " do you mean all of the logic in the for loop?  I.e. move the logic in the for loop into a `protected virtual void ChangeTypeScope(TypeReference tr)` ?